src/js/components/Tabs/Tabs.js

```diff
@@ -93,37 +185,150 @@ const Tabs = forwardRef(
+        tabRefs[tabRefs.length - 1].current.focus();
+        setFocusIndex(tabRefs.length - 1);
+      }
+    };
+
     return (
-      <StyledTabs
-        ref={ref}
-        as={Box}
-        role="tablist"
-        flex={flex}
-        responsive={responsive}
-        {...rest}
-        background={theme.tabs.background}
-      >
-        <StyledTabsHeader
+      <Keyboard onLeft={onLeft} onRight={onRight} onHome={onHome} onEnd={onEnd}>
+        <StyledTabs
+          ref={ref}
           as={Box}
-          direction="row"
-          justify={justify}
-          alignSelf={alignControls}
-          flex={false}
-          wrap
-          background={theme.tabs.header.background}
-          gap={theme.tabs.gap}
-          {...tabsHeaderStyles}
-        >
-          {tabs}
-        </StyledTabsHeader>
-        <StyledTabPanel
+          role="tablist"
           flex={flex}
-          aria-label={tabContentTitle}
-          role="tabpanel"
+          responsive={responsive}
+          {...rest}
+          background={theme.tabs.background}
         >
-          {activeContent}
-        </StyledTabPanel>
-      </StyledTabs>
+          <Box direction={overflow && scroll ? 'row' : 'column'}>
+            {overflow && scroll && (
+              <Button
+                a11yTitle="Previous Tab"
+                icon={<Previous />}
+                disabled={disableLeftArrow}
+                // removed from tabIndex, button is redundant for keyboard users
+                tabIndex={-1}
+                onClick={() => {
+                  let scrolledToIndex;
+                  for (let i = 0; i < tabRefs.length - 1; i += 1) {
+                    if (
+                      !isVisible(tabRefs[i].current) &&
+                      isVisible(tabRefs[i + 1].current)
+                    ) {
+                      if (scroll && scroll.step)
+                        i = Math.max(i - (scroll.step - 1), 0);
+                      scrolledToIndex = i;
+                      tabRefs[i].current.scrollIntoView({ behavior: 'smooth' });
+                      break;
+                    }
+                  }
+
+                  setDisableRightArrow(false);
+                  if (scrolledToIndex === 0) {
+                    // wait for scroll animation to finish
+                    const checkVisible = setInterval(() => {
+                      if (isVisible(tabRefs[0].current)) {
+                        setDisableLeftArrow(true);
+                        clearInterval(checkVisible);
+                      }
+                    }, 100);
+                    setTimeout(() => {
+                      clearInterval(checkVisible);
+                    }, 500);
+                  }
+                }}
+              />
+            )}
+            <StyledTabsHeader
+              ref={headerRef}
+              as={Box}
+              direction="row"
+              justify={justify}
+              alignSelf={alignControls}
+              flex={!!scroll}
```

Comment:
Looks like you were looking at an old version of the file. Currently this is `flex={flex}`. We have a `flex` prop on Tabs so I think we need to leave `flex={flex}` instead of having flex be false or undefined.

Reply:
Oops I was looking at the wrong section of code. So currently this is `flex={!!overflow}` we need flex to be true if the content is overflowing so that the tabs fit between the left and right arrow buttons.